source/android/Renaissance/lib/edit_profile.dart

```diff
@@ -0,0 +1,217 @@
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:portakal/file_converter.dart';
+import 'package:snippet_coder_utils/multi_images_utils.dart';
+
+import 'http_services.dart';
+import 'models/user.dart';
+import 'my_colors.dart';
+
+class EditProfilePage extends StatefulWidget {
+  const EditProfilePage({Key? key, required this.delegate}) : super(key: key);
+  final EditProfileDelegate delegate;
+
+  @override
+  _EditProfilePageState createState() => _EditProfilePageState();
+}
+
+mixin EditProfileDelegate {
+  void onSuccessfulSave();
+}
+
+class _EditProfilePageState extends State<EditProfilePage> {
+  var _image;
+  var profileImg;
+  TextEditingController firstNameController = TextEditingController();
+  TextEditingController lastNameController = TextEditingController();
+  TextEditingController bioController = TextEditingController();
+  bool _isLoading = false;
+  void loadPhoto() async {
+    if (User.me!.photo == null) { return ; }
+    setState(() {
+      _isLoading = true;
+    });
+    profileImg = await FileConverter.getImageFromBase64(User.me!.photo!);
+    setState(() {
+      _isLoading = false;
+    });
+  }
+  @override
+  Widget build(BuildContext context) {
+    if (!_isLoading && profileImg == null) {
+      loadPhoto();
+    }
+    return GestureDetector(
+      onTap: () => FocusManager.instance.primaryFocus?.unfocus(),
+      child: Scaffold(
+        resizeToAvoidBottomInset: false,
+          appBar: AppBar(
+            backgroundColor: MyColors.blue,
+            title: Text("Edit Profile"),
+            centerTitle: true,
+          ),
+          body: CustomScrollView(
+            scrollDirection: Axis.vertical,
+
+            slivers: [SliverFillRemaining(
+              hasScrollBody: false,
+              child: Padding(
+                padding: const EdgeInsets.all(15.0),
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    ClipRRect(
+                      borderRadius: BorderRadius.circular(40.0),
+                      child: _image != null ? Image.file(_image, width: 80, height: 80, fit: BoxFit.fitHeight,) :
+                          User.me!.photo == null ?
+                      CircleAvatar(
+                          backgroundColor: Colors.blueAccent,
+                          child: Text(User.me!.username![0].toUpperCase()),
+                          radius: 32
+                      ) : _isLoading ? CircularProgressIndicator() :
+                       Image.file(profileImg!, width: 80, height: 80, fit: BoxFit.fitHeight,),
```

Comment:
Isn't it better to have the sizes of the profile image responsive. Could be noted for the further implementations in order to prevent compatibility issues on different devices.

Reply:
Noted, will be changed in case of a compatibility problem.